### ISSUE TITLE
Unable to create user with privilege REPLICATION CLIENT on MariaDB 10.5.5

### ISSUE THREAD

**Issue by ghost** [ansible-collections/ansible.mysql#232]
##### SUMMARY
When trying to create a new user using the mysql_user module it error if you try to user REPLICATION CLIENT as a priv. As per the MariaDB docs this was renamed in v10.5.2 to BINLOG MONITOR but the old name should alias to this. In v2.1.0 of community.mysql it works as intended but in v2.3.0 it errors.

It would be nice to be able to run the same Ansible against a MariaDB 10.4 and 10.5 server.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

community.mysql.mysql_user

##### ANSIBLE VERSION

```
ansible [core 2.11.6]
config file = /codebuild/output/src941743992/src/ansible.cfg
configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
executable location = /usr/local/bin/ansible
python version = 3.8.10 (default, Sep 28 2021, 16:10:42) [GCC 9.3.0]
jinja version = 3.0.2
libyaml = True
```

##### COLLECTION VERSION

```
community.mysql 2.3.0
```

##### CONFIGURATION

```

```

##### OS / ENVIRONMENT

OS Ubuntu 20.04.3 LTS
MariaDB 10.5.5

##### STEPS TO REPRODUCE

```yaml
- name: "Create user"
  mysql_user:
    name: "user"
    host: "localhost"
    password: "password"
    login_user: root
    login_password: "password"
    priv: "*.*:SELECT,PROCESS,SHOW DATABASES,SHOW VIEW,EXECUTE,SUPER,CREATE TEMPORARY TABLES,LOCK TABLES,REPLICATION SLAVE,REPLICATION CLIENT,EVENT,TRIGGER,RELOAD"
    state: present
```

##### EXPECTED RESULTS

User created with privileges set as above.

##### ACTUAL RESULTS

Error received as below

```paste below
 TASK [ansible-mariadb : Create user] **************************
fatal: [hostname]: FAILED! => changed=false 
  msg: 'invalid privileges string: Invalid privileges specified: frozenset([''REPLICATION CLIENT''])' 
```

**Comment by tuhoanganh** [comment]
I installed ansible on Macbook Bigsur and Centos 7 few days ago and got the same error with mysql_user module.
However, My ubuntu wsl on window (install ansible 2 months ago) run without any problem.

Note: I'll check and use the exact ansible version to reproduce this error and feedback later

**Comment by Andersson007** [comment]
@jb-2197 @tuhoanganh thanks for the feedback!

https://github.com/ansible-collections/community.mysql/pull/217

~~@sicart any ideas on how we can solve it quickly?~~ @sicart I apologize - copy-paste mistake

**Comment by Andersson007** [comment]
@Jorge-Rodriguez @bmalynovytch should we revert the PR and release a patch release? (we should definitely add mariadb to our CI one day...)

**Comment by Andersson007** [comment]
@rsicart it relates to https://github.com/ansible-collections/community.mysql/pull/217. any ideas on how we can solve it quickly?

**Comment by Andersson007** [comment]
FOR USERS: While we're solving the issue, please install the previous version of the collection using the following command `ansible-galaxy collection install community.mysql:==2.2.0`.
Refer to the [collection guide](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html) for details.

**Comment by tuhoanganh** [comment]
> FOR USERS: While we're solving the issue, please install the previous version of the collection using the following command `ansible-galaxy collection install community.mysql:==2.2.0`. Refer to the [collection guide](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html) for details.

This workaround fixed the problem.
Thank you so much :)

**Comment by Andersson007** [comment]
@tuhoanganh you're welcome!:)

**Comment by Andersson007** [comment]
@bmalynovytch @Jorge-Rodriguez @rsicart one of possible options would be removing that comparison of frozensets from the code.. (and eliminating getting valid privs entirely. Users should know what their servers support).

What the point to check for validity of the privilege name the user passes if the server will through a corresponding exception anyway? Why not just catching it with `try: ... except:..`?
The question is how to do it gracefully in all cases.

What do you think?

**Comment by Andersson007** [comment]
But i believe this is a breaking change and the matter of a long term perspective.

I think there's no way back.

**Option 2 (for now):** could be returning VALID PRIVS and if the exception occurs we could check if the priv is there. I'll create a PR!:)

**Comment by bmalynovytch** [comment]
@Andersson007 I thought this was handled by one of the latests PRs 🤔 

[Edit]
https://github.com/ansible-collections/community.mysql/pull/217

**Comment by Andersson007** [comment]
@bmalynovytch :) please see the description in https://github.com/ansible-collections/community.mysql/pull/233

**Comment by Andersson007** [comment]
I created https://github.com/ansible-collections/community.mysql/pull/233 to fix a bug introduced by https://github.com/ansible-collections/community.mysql/pull/217.

@rsicart @Jorge-Rodriguez also please see https://github.com/ansible-collections/community.mysql/pull/233

**Comment by Andersson007** [comment]
I've just released 2.3.1 and it's already on Galaxy. It contains this fix. Available to install via `ansible-galaxy`. Will get into Ansible main package with its next minor / patch release automatically.

@jb-2197 @tuhoanganh @bmalynovytch thank you!

**Comment by ghost** [comment]
Thanks @Andersson007 and everyone for your swift response on this.
